tag/sqlstore/tag_repository.go

```diff
@@ -0,0 +1,371 @@
+package sqlstore
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/odpf/columbus/tag"
+	"github.com/pkg/errors"
+
+	"gorm.io/gorm"
+)
+
+var (
+	errNilDomainTag    = errors.New("domain tag is nil")
+	errEmptyRecordType = errors.New("record type should not be empty")
+	errEmptyRecordURN  = errors.New("record urn should not be empty")
+)
+
+// TagRepository is a type that manages tag operation ot the primary database
+type TagRepository struct {
+	dbClient *gorm.DB
+}
+
+// Create inserts tag to database
+func (r *TagRepository) Create(domainTag *tag.Tag) error {
+	if r.dbClient == nil {
+		return errNilDBClient
+	}
+	if domainTag == nil {
+		return errNilDomainTag
+	}
+	recordModelTemplate, err := r.getTemplateWithFields(domainTag.TemplateURN)
+	if err != nil {
+		return err
+	}
+	domainTemplate := r.convertToDomainTemplate(recordModelTemplate)
+	err = r.dbClient.Transaction(func(tx *gorm.DB) error {
+		return r.insertDomainTagToDB(tx, domainTag)
+	})
+	if err == nil {
```

Comment:
We could check err! =nil first here right? To avoid nested if

Reply:
yeah I guess we could, the logic is directly from dexter, if the test is not breaking will apply the changes.